backends/arm/test/ops/test_conv.py

```diff
@@ -0,0 +1,330 @@
+                    padding=padding[i],
+                    dilation=dilation[i],
+                    groups=groups[i],
+                    bias=bias[i],
+                    padding_mode=padding_mode[i],
+                ).to(dtype),
+            )
+
+    def get_inputs(self):
+        return self.inputs
+
+    def forward(self, x):
+        for i in range(self.nbr_convs):
+            conv = getattr(self, f"conv_{i}")
+            x = conv(x)
+        return x
+
+
+conv2d_2x2_3x1x40x40_nobias = Conv2d(
+    in_channels=1,
+    out_channels=3,
+    kernel_size=(2, 2),
+    stride=1,
+    bias=False,
+    padding=0,
+    width=40,
+    height=40,
+    batches=3,
+)
+
+conv2d_3x3_1x3x256x256_st1 = Conv2d(
+    in_channels=3,
+    out_channels=10,
+    kernel_size=(3, 3),
+    stride=1,
+    padding=0,
+    width=256,
+    height=256,
+    batches=1,
+)
+
+conv2d_3x3_1x3x12x12_st2_pd1 = Conv2d(
+    in_channels=3,
+    out_channels=4,
+    kernel_size=(3, 3),
+    stride=2,
+    padding=1,
+    width=12,
+    height=12,
+    batches=1,
+)
+
+conv2d_1x1_1x2x128x128_st1 = Conv2d(
+    in_channels=2,
+    out_channels=1,
+    kernel_size=(1, 1),
+    stride=1,
+    padding=0,
+    width=128,
+    height=128,
+    batches=1,
+)
+
+conv2d_2x2_1x1x14x14_st2 = Conv2d(
+    in_channels=1,
+    out_channels=1,
+    kernel_size=(2, 2),
+    stride=2,
+    padding=0,
+    width=14,
+    height=14,
+    batches=1,
+)
+
+conv2d_5x5_3x2x128x128_st1 = Conv2d(
+    in_channels=2,
+    out_channels=3,
+    kernel_size=(5, 5),
+    stride=1,
+    padding=0,
+    width=128,
+    height=128,
+    batches=3,
+)
+
+conv2d_3x3_1x3x224x224_st2_pd1 = Conv2d(
+    in_channels=3,
+    out_channels=16,
+    kernel_size=(3, 3),
+    stride=2,
+    padding=1,
+    width=224,
+    height=224,
+    batches=1,
+)
+
+two_conv2d_nobias = Conv2d(
+    nbr_conv=2,
+    width=256,
+    height=256,
+    in_channels=[3, 10],
+    out_channels=[10, 15],
+    kernel_size=[(5, 5), (5, 5)],
+    stride=[1, 1],
+    padding=[0, 0],
+    bias=[False, False],
+    batches=1,
+)
+
+two_conv2d = Conv2d(
+    nbr_conv=2,
+    width=256,
+    height=256,
+    in_channels=[3, 10],
+    out_channels=[10, 15],
+    kernel_size=[(5, 5), (5, 5)],
+    stride=[1, 1],
+    padding=[0, 0],
+    bias=[True, True],
+    batches=1,
+)
+
+# Shenanigan to get a nicer output when test fails. With unittest it looks like:
+# FAIL: test_conv2d_tosa_BI_2_3x3_1x3x12x12_st2_pd1
+testsuite = [
+    ("2x2_3x1x40x40_nobias", conv2d_2x2_3x1x40x40_nobias),
+    ("3x3_1x3x256x256_st1", conv2d_3x3_1x3x256x256_st1),
+    ("3x3_1x3x12x12_st2_pd1", conv2d_3x3_1x3x12x12_st2_pd1),
+    ("1x1_1x2x128x128_st1", conv2d_1x1_1x2x128x128_st1),
+    ("2x2_1x1x14x14_st2", conv2d_2x2_1x1x14x14_st2),
+    ("5x5_3x2x128x128_st1", conv2d_5x5_3x2x128x128_st1),
+    ("3x3_1x3x224x224_st2_pd1", conv2d_3x3_1x3x224x224_st2_pd1),
+    ("two_conv2d_nobias", two_conv2d_nobias),
+    ("two_conv2d", two_conv2d),
```

Comment:
Do we also want to add memory format tests? i.e. if we can correctly handle `nchw` and `nchw` or not through conversions. cc @Jerry-Ge

Reply:
question: all those conv2d tests won't handle `nchw` in TOSA, so what's the purpose of adding this test?

Reply:
True, it won't be handled in TOSA. This PR is mainly porting unit tests to the new test structure (and adding some test width).

Reply:
OK